interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/KotlinPoetMetadata.kt

```diff
@@ -82,14 +99,16 @@ public inline fun <reified T : KotlinClassMetadata> Metadata.toKotlinClassMetada
  * Returns the [KotlinClassMetadata] this represents. In general you should only use this function
  * when you don't know what the underlying [KotlinClassMetadata] subtype is, otherwise you should
  * use one of the more direct functions like [toKmClass].
+ *
+ * @param lenient see docs on [KotlinClassMetadata.readStrict] and [KotlinClassMetadata.readLenient] for more details.
  */
 @KotlinPoetMetadataPreview
-public fun Metadata.readKotlinClassMetadata(): KotlinClassMetadata {
-  val metadata = KotlinClassMetadata.readStrict(asClassHeader())
-  checkNotNull(metadata) {
-    "Could not parse metadata! Try bumping kotlinpoet and/or kotlinx-metadata version."
+public fun Metadata.readKotlinClassMetadata(lenient: Boolean): KotlinClassMetadata {
+  return if (lenient) {
+    KotlinClassMetadata.readLenient(asClassHeader())
+  } else {
+    KotlinClassMetadata.readStrict(asClassHeader())
```

Comment:
Does `asClassHeader()` do anything? It looks like it's copying from `Metadata` to `Metadata` (an identity transform?).

Reply:
Yep leftover from days past and probably the 0.9.0 merge with this branch. Let me clean up

Reply:
[`c7a7df8` (#1766)](https://github.com/square/kotlinpoet/pull/1766/commits/c7a7df86caa614ea1899e00a0ff0718796967a47)